SwiftGit2/Repository.swift

```diff
@@ -558,4 +579,234 @@ final public class Repository {
 		let iterator = CommitIterator(repo: self, root: branch.oid.oid)
 		return iterator
 	}
+
+	// MARK: - Diffs
+
+	public func diff(for commit: Commit) -> Result<Diff, NSError> {
+		typealias Delta = Diff.Delta
+
+		guard !commit.parents.isEmpty else {
+			// Initial commit in a repository
+			return self.diff(from: nil, to: commit.oid)
+		}
+
+		var mergeDiff: OpaquePointer? = nil
+		for parent in commit.parents {
+			let error = self.diff(from: parent.oid, to: commit.oid) { (diff: Result<OpaquePointer, NSError>) -> NSError? in
+				guard diff.error == nil else {
+					return diff.error!
+				}
+
+				if mergeDiff == nil {
+					mergeDiff = diff.value!
+				} else {
+					let mergeResult = git_diff_merge(mergeDiff, diff.value)
+					guard mergeResult == GIT_OK.rawValue else {
+						return NSError(gitError: mergeResult, pointOfFailure: "git_diff_merge")
+					}
+				}
+				return nil
+			}
+
+			if error != nil {
+				return Result<Diff, NSError>.failure(error!)
+			}
+		}
+
+		defer { git_object_free(mergeDiff) }
```

Comment:
This isn't freed in either of the error conditions here.

Let's move this `defer` to be immediately after the `var mergeDiff` declaration.

Reply:
good catch :+1: